Library/OcTextInputLib/OcTextInputLib.c

```diff
@@ -0,0 +1,526 @@
+  { 0x1D, DEBUG_POINTER ("CTRL+]"),  DEBUG_POINTER ("GroupSeparator")            },
+  { 0x1E, DEBUG_POINTER ("CTRL+^"),  DEBUG_POINTER ("RecordSeparator")           },
+  { 0x1F, DEBUG_POINTER ("CTRL+_"),  DEBUG_POINTER ("UnitSeparator")             }
+};
+
+// Global variables for compatibility protocol
+STATIC EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL  *mOriginalSimpleTextInputEx = NULL;
+STATIC EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL  mCompatSimpleTextInputEx;
+STATIC BOOLEAN                            mProtocolInstalled = FALSE;
+
+/**
+   Compatibility implementation of ReadKeyStrokeEx that falls back to ReadKeyStroke.
+
+   On EFI 1.1 systems like cMP5,1, the SimpleTextInputEx protocol is not available,
+   and CTRL key combinations are handled differently. This function provides:
+
+   1. SimpleTextInputEx protocol compatibility
+   2. Key remapping for enhanced cMP5,1 compatibility:
+      - F10 → CTRL+E (Help functionality for programs that use CTRL+E)
+
+   ESC key is left unchanged as most programs expect ESC to work normally.
+   This allows programs to work with F10 as an alternative to CTRL+E on systems
+   where CTRL combinations may not be reliable.
+
+   @param  This                 Protocol instance pointer.
+   @param  KeyData              A pointer to a buffer that is filled in with the keystroke
+                               state data for the key that was pressed.
+
+   @retval EFI_SUCCESS          The keystroke information was returned.
+   @retval EFI_NOT_READY        There was no keystroke data available.
+   @retval EFI_DEVICE_ERROR     The keystroke information was not returned due to
+                               hardware errors.
+ **/
+
+/**
+  Shared function to read a key stroke from text input with EFI 1.1 compatibility.
+
+  This function provides the common key processing logic that can be shared
+  between the library and driver implementations.
+
+  @param  KeyData         Pointer to a buffer to fill with the key data.
+
+  @retval EFI_SUCCESS     The key stroke was successfully read.
+  @retval EFI_NOT_READY   There was no key stroke data available.
+  @retval EFI_INVALID_PARAMETER KeyData is NULL.
+**/
+EFI_STATUS
+EFIAPI
+OcReadKeyStrokeFromTextInput (
+  OUT EFI_KEY_DATA  *KeyData
+  )
+{
+  EFI_STATUS  Status;
+
+  if (KeyData == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  // Clear the KeyData structure
+  ZeroMem (KeyData, sizeof (EFI_KEY_DATA));
+
+  // Try to read from the standard SimpleTextInput protocol
+  Status = gST->ConIn->ReadKeyStroke (gST->ConIn, &KeyData->Key);
+
+  if (!EFI_ERROR (Status)) {
+    // Handle F10 key remapping before shared processing
+    if (KeyData->Key.ScanCode == SCAN_F10) {
+      DEBUG ((DEBUG_INFO, "OcTextInputLib: F10 key detected - remapping to CTRL+E (Help)\n"));
+      // Remap F10 to CTRL+E for programs that expect CTRL+E for help
+      KeyData->Key.ScanCode    = SCAN_NULL;
+      KeyData->Key.UnicodeChar = 5; // CTRL+E
```

Comment:
In terms of mapping F10 to CTRL+E, I can enter the help menu using this, but I cannot exit it again - is this working for you?

Also if I understand correctly there is no support for CTRL chars in this code, the aim is to operate the editor only with fn. keys, is that correct?